lib/libutee/include/tee_api_defines.h

```diff
@@ -331,12 +331,25 @@
 #define __OPTEE_ALG_ECDH_P384			0x80004042
 #define __OPTEE_ALG_ECDH_P521			0x80005042
 
-/* TEE_ALG_ECDSA_P* and TEE_ALG_ECDH_P* are deprecated */
-#define TEE_ALG_ECDSA_P192			TEE_ALG_ECDSA_SHA1
-#define TEE_ALG_ECDSA_P224			TEE_ALG_ECDSA_SHA224
-#define TEE_ALG_ECDSA_P256			TEE_ALG_ECDSA_SHA256
-#define TEE_ALG_ECDSA_P384			TEE_ALG_ECDSA_SHA384
-#define TEE_ALG_ECDSA_P521			TEE_ALG_ECDSA_SHA512
+/*
+ * TEE_ALG_ECDSA_P* are deprecated on API v1.2 or later and are replaced
+ * with TEE_ALG_ECDSA_SHA* and that have the same value but a different
+ * meaning (defined the related hash algorithm). Therefore assign ID values
+ * of __OPTEE_ALG_ECDSA_P* to the deprecated macros to avoid confusion
+ * while still supporting the algorithm for portability and backward
+ * compatibility purposes.
+ */
+#define TEE_ALG_ECDSA_P192			__OPTEE_ALG_ECDSA_P192
```

Comment:
The previous code tried to phase out the old values defined by __OPTEE_ALG_ECDSA_P* so if a TA is recompiled it will use the new values. The day when we finally say good riddance to __OPTEE_ALG_ECDSA_P* in OP-TEE core only the old TAs that never were recompiled will stop working. With this change, things become less future-compatible.

Reply:
I guess then I need to change management of all `TEE_ALG_ECDSA_Px` to allow any key size and change management of  `TEE_ALG_ECDSA_SHA*` to allow any hash size.

Reply:
All the `__OPTEE_ALG_ECDSA_P*` are already translated into `TEE_ALG_ECDSA_SHA*` in OP-TEE core before it's used. Why do you need to change `TEE_ALG_ECDSA_SHA*` to allow any hash size?

Reply:
If `TEE_ALG_ECDSA_P*` is not expected to enforce the hash size, having `TEE_ALG_ECDSA_SHAx=TEE_ALG_ECDSA_Py` will not allow extracting the hash algo from `TEE_ALG_ECDSA_SHA*`, nor enforcing `TEE_ALG_ECDSA_Py` targets a y-bit key.
Or maybe we can explicitly mention that each `TEE_ALG_ECDSA_P*` enforces a specific a specific hash algo and does not put constraint on the effective key size.